modules/stochastic_tools/src/transfers/SamplerPostprocessorTransfer.C

```diff
@@ -83,27 +120,34 @@ SamplerPostprocessorTransfer::executeFromMultiapp()
     if (_multi_app->hasLocalApp(i))
     {
       FEProblemBase & app_problem = _multi_app->appProblemBase(i);
-      _current_data.emplace_back(app_problem.getPostprocessorValue(_sub_pp_name));
+      for (std::size_t j = 0; j < _sub_pp_names.size(); ++j)
+        _current_data[j].emplace_back(app_problem.getPostprocessorValue(_sub_pp_names[j]));
     }
   }
 }
 
 void
 SamplerPostprocessorTransfer::finalizeFromMultiapp()
 {
-  _results->setCurrentLocalVectorPostprocessorValue(_sampler_ptr->name(), std::move(_current_data));
-  _current_data.clear();
+  for (std::size_t j = 0; j < _sub_pp_names.size(); ++j)
```

Comment:
`MooseIndex`

Reply:
This is another sore point for me. I really don't think we should be using a macro in for loops just because we forget to use size_t. for loops are one of the most fundamental parts of a language and we go and mess it up with a macro which one of the more obscure parts.

Reply:
Fair enough if you know the comparator will be another std::size_t.